openbas-api/src/main/java/io/openbas/service/ChallengeService.java

```diff
@@ -124,57 +139,58 @@ public ChallengesReader playerChallenges(String exerciseId, User user) {
   public ChallengesReader validateChallenge(
       String exerciseId, String challengeId, ChallengeTryInput input, User user) {
     ChallengeResult challengeResult = tryChallenge(challengeId, input);
-    if (challengeResult.getResult()) {
+    if (challengeResult.isResult()) {
       // Find and update the expectations linked to the user
       List<InjectExpectation> playerExpectations =
           injectExpectationRepository.findByUserAndExerciseAndChallenge(
               user.getId(), exerciseId, challengeId);
       playerExpectations.forEach(
           playerExpectation -> {
-            playerExpectation.setScore(playerExpectation.getExpectedScore());
-            InjectExpectationResult expectationResult =
-                InjectExpectationResult.builder()
-                    .sourceId("challenge")
-                    .sourceType("challenge")
-                    .sourceName("Challenge validation")
-                    .result(Instant.now().toString())
-                    .date(Instant.now().toString())
-                    .score(playerExpectation.getExpectedScore())
-                    .build();
-            playerExpectation.getResults().add(expectationResult);
-            playerExpectation.setUpdatedAt(Instant.now());
-            injectExpectationRepository.save(playerExpectation);
+            InjectStatus injectStatus = playerExpectation.getInject().getStatus().orElseThrow();
```

Comment:
Even if this should never happen, could you add a descriptive exception to the orElseThrow?